store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java

```diff
@@ -57,24 +64,48 @@
  */
 public class FederatedStore extends Store {
 
+    /**
+     * Separator used when the prefix of the federated store is added to a
+     * graph ID.
+     */
+    public static final String PREFIX_SEPARATOR = "_";
+
     // Default graph IDs to execute on
     private List<String> defaultGraphIds = new LinkedList<>();
 
     // Cached list of available graphs
     private final List<GraphSerialisable> graphs = new LinkedList<>();
 
+    // Store specific handlers
+    public final Map<Class<? extends Operation>, OperationHandler<?>> storeHandlers = Stream.of(
+            new SimpleEntry<>(AddGraph.class, new AddGraphHandler()),
+            new SimpleEntry<>(GetSchema.class, new GetSchemaHandler()))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
     /**
      * Add a new graph so that it is available to this federated store.
+     * Note the graph ID of the added graph will have the graph ID of
+     * this federated store prefixed to it so it will be identifiable,
+     * and to avoid collisions with other stores.
      *
      * @param graph The serialisable instance of the graph.
      */
     public void addGraph(final GraphSerialisable graph) {
-        graphs.add(graph);
+        // Prefix this store's graph ID to the graph to be added
```

Comment:
Does this mean we won't be able to point to existing accumulo tables?

Reply:
True, come to think of it this is probably not a required feature I will remove.